src/pipdeptree/_models/package.py

```diff
@@ -203,13 +213,17 @@ def render_as_root(self, *, frozen: bool) -> str:
     def render_as_branch(self, *, frozen: bool) -> str:
         if not frozen:
             req_ver = self.version_spec if self.version_spec else "Any"
-            return f"{self.project_name} [required: {req_ver}, installed: {self.installed_version}]"
+            return f"{self.project_name} [required: {req_ver}, extra: {self.req_extra}, installed: {self.installed_version}]"
```

Comment:
As I said, I think we shouldn't overcomplicate this PR and figure out how to handle extras in some other PR.

Reply:
Now your example is starting to make sense. Looking CT3's [extras](https://github.com/CheetahTemplate3/cheetah3/blob/master/SetupConfig.py#L140-L143), it appears that they have the 'filters' and 'markdown' extras that both just contain a package called `markdown`. `Distribution.requires` returns a list of requirement strings.

From reading `packaging`'s [docs](https://packaging.pypa.io/en/stable/requirements.html), it appears that it can parse these requirement strings. What I believe is happening is `Distribution.requires()` is being given these requirement strings where some may represent the same exact package. Therefore, `PackageDAG.from_pkgs()` may create duplicate `ReqPackage` objects (i.e. we encounter the same package names multiple times) which in turn is why we are seeing your example.